packages/core/tests/wallet.test.ts

```diff
@@ -157,4 +157,27 @@ describe('wallet', () => {
     // Expect same basic message record to exist in new wallet
     expect(await bobBasicMessageRepository.getById(basicMessageRecord.id)).toMatchObject(basicMessageRecord)
   })
+
+  test('changing wallet key', async () => {
```

Comment:
This is using the same key to rekey the wallet. Would be good to add a test that does the following:

- create and open the wallet
- close the wallet
- rotate the wallet key to a new key
- open the wallet (using the agent.wallet so you can provide the new key directly -> more explicit)